File: source/example/plswigsimp/testplswigsimp.cs
// This example illustrates how C++ interfaces can be used from C#.

using System;
using example.plswigsimp;

public class testplswigsimp
{
    static void Main()
    {
        // ----- Object creation -----
        Console.WriteLine( "Hello testplswigsimp in C#." );
        
        Simple a = new Simple("");
        Console.WriteLine( "Created a object of Simple: " + a.valid() + ", " + a);
        
        Simple2 b = new Simple2(a.p());
        Console.WriteLine( "Converted a interface: " + (b.valid() ? "ok, " : " fail, ") + b);
        
        // ----- Call some methods -----
        
        Console.WriteLine( "Simple.add(1, 2) = " + a.add(1, 2) );
        
        Console.WriteLine( "Goodbye" );
    }
}

File: source/example/plswigsimp/testplswigsimp.pl
# This example illustrates how C++ interfaces can be used from Perl.

use plswigsimp;

print "Hello testplswigsimp in Perl.\n";

$a = new plswigsimp::Simple("");
print "Created a object of Simple: $a\n";

$b = new plswigsimp::Simple2($a->p());
print "Converted a interface: $b\n";

print "Simple.add(1, 2) = ", $a->add(1, 2), "\n";
